### PR TITLE
Fixes Holoparasites killing their hosts when suiciding.

### LINKED ISSUE #7389: Suiciding as a holoparasite kills the host
## Description

Using `/suicide` as a holoparasite kills the host too.

Using `/ghost` removes the player and reregisters the ghost role so that someone else can take it. So I suggest either preventing `/suicide` on holoparasites or redirecting the command to run `/ghost` instead.

### PR DESCRIPTION
**Pull request by keronshb**
## About the PR 

This fixes #7389 so any Guardian type that /suicides will now /ghost instead.

It also adds a new tag to help for these kinds of situations.

A huge thank you to @moonheart08 for pointing me in the right direction.

**Screenshots**

**Changelog**

:cl:
- add: Holoparasites no longer kill their hosts when suiciding.

### CODE REVIEW

File: Content.Server/Chat/Commands/SuicideCommand.cs [hunk 1]
```diff
@@ -82,6 +83,13 @@ public void Execute(IConsoleShell shell, string argStr, string[] args)
                 return;
             }
 
+            //Checks to see if the CannotSuicide tag exits, ghosts instead.
+            if(EntitySystem.Get<TagSystem>().HasTag(owner, "CannotSuicide"))
+            {
+                EntitySystem.Get<GameTicker>().OnGhostAttempt(mind, false);
```
Comment (moonheart08) [inline Content.Server/Chat/Commands/SuicideCommand.cs:89]: This should give an error if ghosting isn't possible, ala the ghost command.

File: Content.Server/Guardian/GuardianSystem.cs [hunk 2]
```diff
@@ -240,6 +240,8 @@ private void OnGuardianDamaged(EntityUid uid, GuardianComponent component, Damag
         {
             if (args.DamageDelta == null) return;
 
+
+
```
Comment (moonheart08) [inline Content.Server/Guardian/GuardianSystem.cs:244]: Huh?
```suggestion
```

**Review by moonheart08 — CHANGES_REQUESTED** [review]

File: Content.Server/Guardian/GuardianSystem.cs (hunk 2 above)
Comment (keronshb) [reply Content.Server/Guardian/GuardianSystem.cs]: Oh oops this was from when I was trying something else

File: Content.Server/Chat/Commands/SuicideCommand.cs (hunk 1 above)
Comment (keronshb) [reply Content.Server/Chat/Commands/SuicideCommand.cs]: Done

File: Content.Server/Chat/Commands/SuicideCommand.cs [hunk 3]
```diff
@@ -86,8 +86,16 @@ public void Execute(IConsoleShell shell, string argStr, string[] args)
             //Checks to see if the CannotSuicide tag exits, ghosts instead.
             if(EntitySystem.Get<TagSystem>().HasTag(owner, "CannotSuicide"))
             {
-                EntitySystem.Get<GameTicker>().OnGhostAttempt(mind, false);
-                return;
+                if (!EntitySystem.Get<GameTicker>().OnGhostAttempt(mind, true))
```
Comment (moonheart08) [inline Content.Server/Chat/Commands/SuicideCommand.cs:89]: This function has side effects, it can only be run once, as it actually performs the ghosting.

**Review by moonheart08 — CHANGES_REQUESTED** [review]

File: Content.Server/Chat/Commands/SuicideCommand.cs (hunk 3 above)
Comment (keronshb) [reply Content.Server/Chat/Commands/SuicideCommand.cs:89]: Should be fixed now

**Pull request merged**